contracts/mocks/MockGate.sol

```diff
@@ -0,0 +1,142 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
```

Comment:
This seems very prone to error, not sure what is going on here ... 
       
<img width="1651" alt="Screenshot 2021-09-23 at 22 43 40" src="https://user-images.githubusercontent.com/699036/134588204-58c15678-3751-4258-8b28-f93626c3bfa6.png">
0

Reply:
Contracts are actually different.
In mockGate, method `registerVoucherSetID` has modifier `onlyFromRouter` instead of `onlyOwner` to test an alternative path for registering voucher set id.

Perhaps you did not have the latest version? This is bidyut's work and when I merged it to this branch, I'm sure it was ok.